### PR TITLE
Adding "create table as" for greenplum dialect

### PR DESCRIPTION
**Pull request by bpfaust**
### Brief summary of the change made

Attempts to add the "create table as" syntax for greenplum based on the same syntax from postgres.  Main addition is the DistributedBySegment grammar which should be at the end of creation statements within greenplum.

Also added support for non-boolean table parameters (ie. compression algorithm names)

### Are there any other side effects of this change that we should be aware of?
None - should be self-contained to the new greenplum dialect.
yaml test generator is failing due to parse errors - would appreciate help in determining why

### Pull Request checklist
- [ ] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_greenplum.py [hunk 1]
```diff
@@ -167,13 +248,77 @@ class CreateTableStatementSegment(postgres.CreateTableStatementSegment):
...
+        ),
+        "AS",
+        OneOf(
+            OptionallyBracketed(Ref("SelectableGrammar")),
+            OptionallyBracketed(Sequence("TABLE", Ref("TableReferenceSegment"))),
+            Ref("ValuesClauseSegment"),
+            OptionallyBracketed(Sequence("EXECUTE", Ref("FunctionSegment"))),
         ),
+        Ref("WithDataClauseSegment", optional=True),
+        Ref("DistributedBySegment"),
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_greenplum.py:323]: this is probably your issue - this wants to be optional

File: src/sqlfluff/dialects/dialect_greenplum.py [hunk 2]
```diff
@@ -167,13 +248,77 @@ class CreateTableStatementSegment(postgres.CreateTableStatementSegment):
...
+                Ref("TemporaryGrammar"),
+            ),
+            "UNLOGGED",
+            optional=True,
+        ),
+        "TABLE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("TableReferenceSegment"),
+        AnyNumberOf(
+            Sequence(
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_greenplum.py:278]: You don't need this Sequence(), you can stick the optional=True in Bracketed()

Comment (bpfaust) [reply src/sqlfluff/dialects/dialect_greenplum.py:278]: Ok - this was copied straight from the postgres dialect file so I left it alone

**Comment by alanmcruickshank** [comment]
I've had a play around with this PR and the problem is that the `distributed_by` is being claimed by the `StatementSegment`. While you've updated the various terminator grammars, unfortunately the one you'd need isn't a grammar 🤦 (no reason you should know that either).

This is going to be easier once I've merged #5186. I'll give you a hand here once that's merged.

**Comment by alanmcruickshank** [comment]
@bpfaust - I've just merged all the recent changes in to see how the tests behave. To see what I meant in the previous comment, check out how the `postgres` grammar treats the `SelectStatementSegment` and how it changes the `terminators` on it. We'll probably need to do something similar within this dialect to also add the new `DISTRIBUTED BY` segment as a terminator to that too.

File: src/sqlfluff/dialects/dialect_greenplum.py (hunk 1 above)
Comment (alanmcruickshank) [reply src/sqlfluff/dialects/dialect_greenplum.py]: I sorted this.

File: src/sqlfluff/dialects/dialect_greenplum.py (hunk 2 above)
Comment (alanmcruickshank) [reply src/sqlfluff/dialects/dialect_greenplum.py]: And sorted this.

**Review by alanmcruickshank — APPROVED** [review]
I realise I've made a bunch of changes here, but I think we're otherwise good to go.

**Pull request merged**